### PR TITLE
Add ReturnTypeWillChange attribute to prevent from deprecation errors on PHP 8.1

### PR DESCRIPTION
**Pull request by Stevad**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    <li>Update mustache template for model classes - added ReturnTypeWillChange attribute to prevent deprecation errors on PHP 8.1</li>
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

Update mustache template for model classes - added ReturnTypeWillChange attribute to prevent deprecation errors on PHP 8.1.

### CODE REVIEW

**Comment by Stevad** [comment]
Related issue: https://github.com/amazon-php/sp-api-sdk/issues/231

**Comment by norberttech** [comment]
hey @Stevad 
what you described (reverting changes) is not how this should work. However I wasn't even thinking about using rector in the first place, but to simply update the mustache file of models since this is probably the only place where this occurs, SDKs itself should be fine. I'm not sure its worth using rector, especially because of how weird rector behavior might be (check also other PR with nullable return types)

**Comment by Stevad** [comment]
@norberttech Ok got it. I will try a solution by updating mustache files and will update my PR.

**Comment by Stevad** [comment]
@norberttech I did rollback for Rector configuration and update model_generic.mustache file. Should be ok now.

P.S. Found a little bit strange thing - to process all files I need to execute rector + php-cs-fixer twice. Might be something with my dev env.

**Comment by norberttech** [comment]
@Stevad are you using php 7.4 to generate the code?

**Comment by Stevad** [comment]
Nope. 8.1. But can regenerate using 7.4. I missed something?

вс, 28 авг. 2022 г., 22:19 Norbert Orzechowicz ***@***.***>:

**Comment by norberttech** [comment]
Technically speaking all dependencies and set up is prepared for PHP 7.4, I’m not sure how it will work at 8.1. That's why there is nix shell setup added to repo, to quickly jump into PHP 7.4.
I don't have to use mentioned tools twice, but I'm using 7.4, that's why I though it might be related

**Comment by Stevad** [comment]
The only issue we have on 8.1 - is one mentioned in https://github.com/amazon-php/sp-api-sdk/issues/230. Everything else seems to work as expected.

**Comment by norberttech** [comment]
I’m talking about tools used to generate code, not code itself. SDK is supporting 8.1 and will work fine but rector, cs fixer etc - tools are configured to work with php 7.4 which might explain why you need to generate code twice. Could you please try php7.4 or just reduce this PR to mustache file adjustments and I will regenerate everything?
Thanks!

**Comment by Stevad** [comment]
I'll regenerate on 7.4. This is not a problem.

пн, 29 авг. 2022 г., 10:55 Norbert Orzechowicz ***@***.***>:

**Comment by Stevad** [comment]
@norberttech I synced the branch with the latest changes from `4.x` and regenerated using PHP 7.4. Take a look, please.

P.S. There is nothing significantly changed.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 vendor/
 .openapi-generator
 var/
-.env
\ No newline at end of file
+.env
+.idea/
```
Comment (norberttech) [inline .gitignore]: please use global .gitignore for IDE/System specific files

File: src/AmazonPHP/SellingPartner/Api/AplusContentApi/APlusSDK.php [hunk 2]
```diff
@@ -52,6 +52,8 @@ public function __construct(ClientInterface $client, HttpFactory $requestFactory
      *
      * @throws ApiException on non-2xx response
      * @throws InvalidArgumentException
+     *
+     * @return \AmazonPHP\SellingPartner\Model\APlus\PostContentDocumentResponse
```
Comment (norberttech) [inline src/AmazonPHP/SellingPartner/Api/AplusContentApi/APlusSDK.php:56]: not sure why this was added 🤔  but lets merge it as is and I will debug it later as it has no impact on the code

Comment (Stevad) [reply src/AmazonPHP/SellingPartner/Api/AplusContentApi/APlusSDK.php:56]: This is after regeneration. Not added manually.

Comment (norberttech) [reply src/AmazonPHP/SellingPartner/Api/AplusContentApi/APlusSDK.php:56]: I know, probably some rules are overlapping between rector and CS fixer, need to debug it, thanks!

**Comment by norberttech** [comment]
thank you very much @Stevad 
this for sure will reduce number of warnings in some projects! 🥳

**Pull request merged**